manifests/init.pp

```diff
@@ -137,4 +140,12 @@
   if $manage_logind {
     contain systemd::logind
   }
+
+  if $dropin_files {
+    $dropin_files.each |$name, $resource| {
+      systemd::dropin_file { $name:
+        * => $resource
+      }
+    }
+  }
```

Comment:
If the Hash is not optional but defaults to `{}`, then you don't need the condition anymore.
```suggestion

  $dropin_files.each |$name, $resource| {
    systemd::dropin_file { $name:
      * => $resource,
    }
  }
```

Ideally even, it'd be good to have a data type for that, like it was done for `$journald_settings` and `$logind_settings` above.